PyRDF/Operation.py

```diff
@@ -88,42 +89,8 @@ def __init__(self, name, *args, **kwargs):
     def _classify_operation(self, name):
         # Classifies the given operation as action or
         # transformation and returns the type.
-
-        future_ops = [
-        "Range",
-        "Take",
-        "Snapshot",
-        "Foreach"
-        ]
-
-        if name in future_ops:
-            raise NotImplementedError("This operation ({}) isn't supported yet ! ".format(name))
-
-        ops = Operation.Types
-
-        operations_dict = {
-        'Define':ops.TRANSFORMATION,
-        'Filter':ops.TRANSFORMATION,
-        'Histo1D':ops.ACTION,
-        'Histo2D':ops.ACTION,
-        'Histo3D':ops.ACTION,
-        'Profile1D':ops.ACTION,
-        'Profile2D':ops.ACTION,
-        'Profile3D':ops.ACTION,
-        'Count':ops.ACTION,
-        'Min':ops.ACTION,
-        'Max':ops.ACTION,
-        'Mean':ops.ACTION,
-        'Sum':ops.ACTION,
-        'Fill':ops.ACTION,
-        'Report':ops.ACTION
-        }
-        
-        op_type = operations_dict.get(name)
-
-        if not op_type:
-            raise Exception("Invalid operation \"{}\"".format(name))
-        return op_type
+        from . import current_backend
```

Comment:
I think these two lines would not even be necessary. `Operation` should stay as a class that contains **all**  the operations and their types, it does not need to care whether the operation is supported in a backend or not, it should not contact the backend. That check should be done outside of `Operation`: first you ask the backend if it supports that operation, and then you ask `Operation` to classify that operation for you, in two steps.